_rules/SVG-image-has-name.md

```diff
@@ -0,0 +1,201 @@
+
+### Expectation
+
+Each target element has an accessible name.
+
+## Assumptions
+
+*There are currently no assumptions*
+
+## Accessibility support
+
+Browser and assistive technology support for SVG `title` and `desc` elements is currently inconsistent. Using WAI ARIA in combination with the `img` role for non-decorative `svg` elements significantly improves accessibility browser support.
+
+## Background
+
+- [Understanding Success Criterion 1.1.1: Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
+- [SVG Accessibility API Mappings](https://www.w3.org/TR/svg-aam-1.0)
+- [WAI-ARIA Graphics Module](https://www.w3.org/TR/graphics-aria-1.0/)
+- [ARIA4: Using a WAI-ARIA role to expose the role of a user interface component](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA4)
+- [ARIA6: Using aria-label to provide labels for objects](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA6)
+- [ARIA10: Using aria-labelledby to provide a text alternative for non-text content](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA10)
+
+## Test Cases
+
+### Passed
+
+#### Pass example 1
+
+The `<svg>` element has a role of `img` which takes its accessible name from its `<title>` element.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+  <title>A yellow circle</title>
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Pass example 2
+
+The `<svg>` element has a role of `img` and an `aria-label` containing the accessible name.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" role="img" width="100" height="100" aria-label="A yellow circle">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Pass example 3
+
+The `<svg>` element has the role of `graphics-document` which takes its accessible name from its' child content, the `<title>` and `<desc>` elements.
+
+```html
+
+<svg xmlns="http://www.w3.org/2000/svg" role="graphics-document">
+    <title>3 circles</title>
+    <desc>3 yellow circles with a green stroke horizontally adjacent to each other
+    </desc>
+    <g role="group">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle cx="150" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle cx="250" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+    </g>
+</svg>
+```
+
+#### Pass example 4
+
+The `<circle>` element is included in the accessibility tree and takes its' accessible name from its' child content, the `<title>` element.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg">
+	<circle role="graphics-symbol" cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow">
+	    <title>Yellow circle</title>
```

Comment:
Questionable: need to check this is valid.
Include test cases from graphics module.

Reply:
In VoiceOver works well. It reads "yellow circle image". If we put [svg][circle ... role="graphics-symbol"][title]title 1[/title][/circle][title]title 2[/title][/svg] we get both titles: "title1 image"; "title 2 group". If we put [svg ... role="graphics-symbol"] or [svg ... role="img"] we get only "title2 image" - doesn't read the title1.

Reply:
Verified applicability and test cases with Leonie Watson and @kasperisager